### DISCUSSION TITLE
Replace Custom Carrier Services

### DISCUSSION THREAD

**Discussion by resistorsoftware** [Shopify/function-examples#233] [API Ideas]
Custom Carrier services are probably the most neglected aspect of Shopify and have remained original since its inception. Even Plus customers have to use them and they are very limited, but super useful as nothing else Shopify offers can replace them. 

Since checkout can be input, and the output would be shipping rates, a Function would be an ideal way to make checkouts superior. As a small ask, I for one, speaking for many, would like to be able to inject a Shipping Rate into the checkout. The only way that is possible is to create an artificially high rate and discount it down, which not only looks stupid but also bugs out Shopify checkout. At least, it bugs out the non-single-page version. 

So how about a general improvement to Functions, and give us a way to dynamically add a shipping rate? Since we can already sort them and show/hide them, this would add a next-level functionality not currently possible, but oh-so-desired.

**Comment by nickwesselman** [comment]
Thanks for the suggestion! Moved to 'API Ideas'.
